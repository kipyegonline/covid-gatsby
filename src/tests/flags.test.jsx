import Enzyme, { shallow, mount } from "enzyme";
import React from "react";
import Adapter from "enzyme-adapter-react-16";
import toJSON, { toJson } from "enzyme-to-json";
import Pagination from "../components/covidreact/countries/pagination";

Enzyme.configure({ adapter: new Adapter() });
describe("<Pagination/>", () => {
  const _click = jest.fn();
  const wrapper = shallow(
    <Pagination pages={10} setPage={_click} current={4} />
  );
  console.log(wrapper.debug());
  test("matches snapshot", () => {
    expect(toJSON(wrapper)).toMatchSnapshot();
  });
  test("presence of number of pages", () => {
    expect(wrapper.find(".page-icon")).toHaveLength(10);
  });
  describe("first element presence", () => {
    const el = wrapper.find(".page-icon").at(1);
    test("presence of second element", () => expect(el.text()).toBe("1"));
    test("click on the el", () => {
      el.simulate("click");
      expect(_click).toHaveBeenCalled();
      expect(_click).toBecalledTimes(2);
      expect(_click).toBeCalledWith(2);
    });
  });
  describe("Current element", () => {
    const elactive = wrapper.find(".page-icon").at(4);
    test("active page", () => {
      expect(elactive.prop("style")).toEqual({ background: "red" });
    });
  });
});
