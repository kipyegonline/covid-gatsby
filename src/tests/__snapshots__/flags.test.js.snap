// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pagination/> matches snapshot 1`] = `
<Fragment>
  <hr
    className="divider"
  />
  <p
    className="text-center"
  >
    Page 
    5
     of 
    10
    .
     
  </p>
  <styled.div>
    <styled.div
      className="page-icon"
      handleClick={[Function]}
      key="0"
      style={
        Object {
          "background": "red",
        }
      }
    >
      1
    </styled.div>
    <styled.div
      className="page-icon"
      handleClick={[Function]}
      key="1"
      style={
        Object {
          "background": "red",
        }
      }
    >
      2
    </styled.div>
    <styled.div
      className="page-icon"
      handleClick={[Function]}
      key="2"
      style={
        Object {
          "background": "red",
        }
      }
    >
      3
    </styled.div>
    <styled.div
      className="page-icon"
      handleClick={[Function]}
      key="3"
      style={
        Object {
          "background": "red",
        }
      }
    >
      4
    </styled.div>
    <styled.div
      className="page-icon"
      handleClick={[Function]}
      key="4"
      style={
        Object {
          "background": "red",
        }
      }
    >
      5
    </styled.div>
    <styled.div
      className="page-icon"
      handleClick={[Function]}
      key="5"
      style={
        Object {
          "background": "gray",
        }
      }
    >
      6
    </styled.div>
    <styled.div
      className="page-icon"
      handleClick={[Function]}
      key="6"
      style={
        Object {
          "background": "gray",
        }
      }
    >
      7
    </styled.div>
    <styled.div
      className="page-icon"
      handleClick={[Function]}
      key="7"
      style={
        Object {
          "background": "gray",
        }
      }
    >
      8
    </styled.div>
    <styled.div
      className="page-icon"
      handleClick={[Function]}
      key="8"
      style={
        Object {
          "background": "gray",
        }
      }
    >
      9
    </styled.div>
    <styled.div
      className="page-icon"
      handleClick={[Function]}
      key="9"
      style={
        Object {
          "background": "gray",
        }
      }
    >
      10
    </styled.div>
  </styled.div>
</Fragment>
`;
